kubetest/process/process_test.go

```diff
@@ -0,0 +1,370 @@
+			}
+			if suite.Failures != 7 {
+				t.Errorf("Case %s interrupted and increase suite failures from 6 to 7, found: %d", tc.name, suite.Failures)
+			}
+		} else {
+			if suite.Failures != 6 {
+				t.Errorf("Case %s passed so suite failures should remain at 6, found: %d", tc.name, suite.Failures)
+			}
+		}
+
+	}
+}
+
+func TestOutput(t *testing.T) {
+	cases := []struct {
+		name              string
+		terminated        bool
+		interrupted       bool
+		causeTermination  bool
+		causeInterruption bool
+		pass              bool
+		sleep             int
+		output            bool
+		shouldError       bool
+		shouldInterrupt   bool
+		shouldTerminate   bool
+	}{
+		{
+			name: "finishRunning can pass",
+			pass: true,
+		},
+		{
+			name:   "output can pass",
+			output: true,
+			pass:   true,
+		},
+		{
+			name:        "finishRuning can fail",
+			pass:        false,
+			shouldError: true,
+		},
+		{
+			name:        "output can fail",
+			pass:        false,
+			output:      true,
+			shouldError: true,
+		},
+		{
+			name:        "finishRunning should error when terminated",
+			terminated:  true,
+			pass:        true,
+			shouldError: true,
+		},
+		{
+			name:        "output should error when terminated",
+			terminated:  true,
+			pass:        true,
+			output:      true,
+			shouldError: true,
+		},
+		{
+			name:              "finishRunning should interrupt when interrupted",
+			pass:              true,
+			sleep:             60,
+			causeInterruption: true,
+			shouldError:       true,
+		},
+		{
+			name:              "output should interrupt when interrupted",
+			pass:              true,
+			sleep:             60,
+			output:            true,
+			causeInterruption: true,
+			shouldError:       true,
+		},
+		{
+			name:             "output should terminate when terminated",
+			pass:             true,
+			sleep:            60,
+			output:           true,
+			causeTermination: true,
+			shouldError:      true,
+		},
+		{
+			name:             "finishRunning should terminate when terminated",
+			pass:             true,
+			sleep:            60,
+			causeTermination: true,
+			shouldError:      true,
+		},
+	}
+
+	clearTimers := func(c *Control) {
+		if !c.Terminate.Stop() {
+			<-c.Terminate.C
+		}
+		if !c.Interrupt.Stop() {
+			<-c.Interrupt.C
+		}
+	}
+
+	for _, tc := range cases {
+		log.Println(tc.name)
+		interrupt := time.NewTimer(time.Duration(0))
+		terminate := time.NewTimer(time.Duration(0))
+		c := NewControl(time.Duration(0), interrupt, terminate, false)
+		c.terminated = tc.terminated
+		c.interrupted = tc.interrupted
+		clearTimers(c)
+		if tc.causeInterruption {
+			interrupt.Reset(0)
+		}
+		if tc.causeTermination {
+			terminate.Reset(0)
+		}
+		var cmd *exec.Cmd
+		if !tc.pass {
+			cmd = exec.Command("false")
+		} else if tc.sleep == 0 {
+			cmd = exec.Command("true")
+		} else {
+			cmd = exec.Command("sleep", strconv.Itoa(tc.sleep))
```

Comment:
@krzyzacy why are we shelling out to `sleep` here?

Reply:
I looked some more, we have our own custom version of cmd.Output essentially so we can timeout the process potentially. I think we can write a more straightforward version but that's beside the point re: solving the new failures here.